exonum-java-binding/common/src/main/java/com/exonum/binding/common/proofs/list/FlatListProof.java

```diff
@@ -109,6 +111,9 @@
           size));
     }
 
+    // Check proof hashes
+    checkProofHashes();
```

Comment:
We can, but that would require duplicating the code in both branches. I don't think it will have a significant impact, as the earlier profile showed that hashing takes most of the time. #1163 could tell the numbers.

Reply:
It is literally noise:
```
(no verification)

Benchmark                      (height)   Mode  Cnt   Score   Error   Units
FlatListProofBenchmark.verify        20  thrpt   10  46,506 ? 0,212  ops/ms

(with verification, flat-list-verify-benchmark-with-hash-checks-ECR-3608)

Benchmark                      (height)   Mode  Cnt   Score   Error   Units
FlatListProofBenchmark.verify        20  thrpt   10  46,407 ? 0,230  ops/ms
```